app/api-v1/controllers/Recipe/index.js

```diff
@@ -1,45 +1,59 @@
 const { runProcess } = require('../../../utils/dscp-api')
-const { client } = require('../../../db')
+const db = require('../../../db')
 const { mapRecipeData } = require('./helpers')
 const { BadRequestError, NotFoundError } = require('../../../utils/errors')
 
-const throwErr = (type, path) => {
+const throwErr = (type, req) => {
   switch (type) {
     case 400:
       throw new BadRequestError({
         message: 'missing parameters',
-        path,
+        req,
       })
     case 404:
       throw new NotFoundError({
         message: 'not found',
-        path,
+        req,
       })
   }
 }
 
 module.exports = {
+  // TODO abstranct to transactions controller and first path e.g. /recipe /order is database model
+  // do this along with order
   transaction: {
+    getAll: async (req) => {
+      const { id } = req.params
+      if (!id) throwErr(400, req)
```

Comment:
won't `req.params` always contain `id` because of open-api validation?

Reply:
This is something we are not 100% sure. I would expect yes, but Matt saying that there might be some issues with the Open API validation so left this check there until we know for sure